src/mame/heathzenith/h89.cpp

```diff
@@ -998,10 +996,94 @@ ROM_START( h89_sigmasoft )
+	ROMX_LOAD("2732_444-142_mtr90.u518",  0x0000, 0x1000, CRC(c4ff47c5) SHA1(d6f3d71ff270a663003ec18a3ed1fa49f627123a), ROM_BIOS(0))
+
+	ROM_SYSTEM_BIOS(1, "mms84b", "MMS 444-84B")
+	ROMX_LOAD("2732_444_84b_mms.u518",    0x0000, 0x1000, CRC(7e75d6f4) SHA1(baf34e036388d1a191197e31f8a93209f04fc58b), ROM_BIOS(1))
+
+	ROM_SYSTEM_BIOS(2, "kmr-100", "Kres KMR-100 V3.a.02")
+	ROMX_LOAD("2732_kmr100_v3_a_02.u518", 0x0000, 0x1000, CRC(fd491592) SHA1(3d5803f95c38b237b07cd230353cd9ddc9858c13), ROM_BIOS(2))
+
+	ROM_SYSTEM_BIOS(3, "mtrhex_4k", "Ultimeth 4k ROM")
+	ROMX_LOAD("2732_mtrhex_4k.u518",      0x0000, 0x1000, CRC(e26b29a9) SHA1(ba13d6c9deef682a9a8262bc910d46b577929a13), ROM_BIOS(3))
+
+	ROM_SYSTEM_BIOS(4, "mtr90-84", "Heath's MTR-90 (444-84 - Superseded by 444-142)")
+	ROMX_LOAD("2732_444-84_mtr90.u518",   0x0000, 0x1000, CRC(f10fca03) SHA1(c4a978153af0f2dfcc9ba05be4c1033d33fee30b), ROM_BIOS(4))
+
+	ROM_SYSTEM_BIOS(5, "mms84a", "MMS 444-84A (Superseded by MMS 444-84B)")
+	ROMX_LOAD("2732_444_84a_mms.u518",    0x0000, 0x1000, CRC(0e541a7e) SHA1(b1deb620fc89c1068e2e663e14be69d1f337a4b9), ROM_BIOS(5))
+ROM_END
+
 } // anonymous namespace
 
 
-//    year  name           parent compat  machine        input class                init        company          fullname                           flags
-COMP( 1979, h88,           h89,   0,      h88,           h88,  h88_state,           empty_init, "Heath Company", "Heathkit H88",                    MACHINE_SUPPORTS_SAVE)
-COMP( 1979, h89,           0,     0,      h89,           h89,  h89_state,           empty_init, "Heath Company", "Heathkit H89",                    MACHINE_SUPPORTS_SAVE)
-COMP( 1984, h89_sigmasoft, h89,   0,      h89_sigmasoft, h89,  h89_sigmasoft_state, empty_init, "Heath Company", "Heathkit H89 with SigmaSoft IGC", MACHINE_SUPPORTS_SAVE)
+//    year  name           parent compat machine        input class                init        company                fullname                   flags
+COMP( 1979, h88,           h89,   0,     h88,           h88,  h88_state,           empty_init, "Heath Company",       "H-88",                    MACHINE_SUPPORTS_SAVE)
+COMP( 1979, h89,           0,     0,     h89,           h89,  h89_state,           empty_init, "Heath Company",       "H-89",                    MACHINE_SUPPORTS_SAVE)
+COMP( 1980, z89,           h89,   0,     h89,           h89,  h89_state,           empty_init, "Zenith Data Systems", "Z-89",                    MACHINE_SUPPORTS_SAVE)
+COMP( 1981, z90,           h89,   0,     h89,           h89,  h89_state,           empty_init, "Zenith Data Systems", "Z-90",                    MACHINE_SUPPORTS_SAVE)
+COMP( 1984, h89_sigmasoft, h89,   0,     h89_sigmasoft, h89,  h89_sigmasoft_state, empty_init, "Heath Company",       "H-89 with SigmaSoft IGC", MACHINE_SUPPORTS_SAVE)
+COMP( 1984, z90_sigmasoft, h89,   0,     h89_sigmasoft, h89,  h89_sigmasoft_state, empty_init, "Zenith Data Systems", "Z-90 with SigmaSoft IGC", MACHINE_SUPPORTS_SAVE)
```

Comment:
Same with this – it adds clutter to the supported system list and so on, and requires dual-maintenance of the copy/pasted ROM definitions.